src/wallet/wallet_rpc_server.cpp

```diff
@@ -1888,6 +1888,13 @@ namespace tools
         rpc_transfers.tx_hash      = epee::string_tools::pod_to_hex(td.m_txid);
         rpc_transfers.subaddr_index = {td.m_subaddr_index.major, td.m_subaddr_index.minor};
         rpc_transfers.key_image    = td.m_key_image_known ? epee::string_tools::pod_to_hex(td.m_key_image) : "";
+        std::vector<std::string> keys;
+        for (auto o : td.m_tx.vout) 
+        {
+          std::string key = epee::string_tools::pod_to_hex(boost::get<cryptonote::txout_to_key>(o.target).key);
```

Comment:
In the similar wallet-cli command it actually only lists one output per key image, as I read, this is listing all the vouts. I believe we can filter this to only be the single vout matching the key image

Reply:
I guess what should be taken care of is to make sure if the get_transfers call is filtered by account.major/minor only the correlating vout/vouts should be returned.